### PR TITLE
Make Builds Environment-Aware

### PR DESCRIPTION
**Pull request by elg0nz**
• GitHub Actions: Added EXTENSION_ENV to manage builds for dev, staging, and production.
• Manifests: Renamed dev manifest to “[Development] Mocksi Lite” and updated production manifest for branding and security.
• Vite Config: Enhanced to pick the right manifest based on EXTENSION_ENV.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new staging manifest for the Chrome extension, enhancing its configuration and permissions.
	- Added a mechanism to select the appropriate manifest based on the environment, improving deployment flexibility.

- **Improvements**
	- Updated content security policies and external connectivity settings for better integration and security.
	- Enhanced the build process with a new environment variable to improve configurability.

- **Branding Changes**
	- Renamed the development version of the application to "[Development] Mocksi Lite" for clarity.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,8 +40,14 @@ jobs:
       - name: Install dependencies
         run: pnpm install --no-frozen-lockfile
 
+        # Set EXTENSION_ENV variable to the current branch name
+      - name: Set EXTENSION_ENV variable
+        run: echo "EXTENSION_ENV=${GITHUB_REF#refs/heads///-/}" >> $GITHUB_ENV
+45:
```
Comment (fitzk) [inline .github/workflows/ci.yml:46]: Screenshot 2024-09-03 at 10 16 03 AM
Screenshot 2024-09-03 at 10 15 50 AM


@elg0nz take a look at this was this intended? I ran the build and zip action and those were the errors ☝🏻

**Review by fitzk — CHANGES_REQUESTED** [review]
everything looks good except the broken build & zip GitHub action is broken, see comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (elg0nz) [reply .github/workflows/ci.yml:46]: Nope, that was a typo :face-palm: correcting it now!

**Comment by fitzk** [comment]
lgtm!

**Review by fitzk — APPROVED** [review]

**Pull request merged**